dask/array/utils.py

```diff
@@ -21,7 +21,16 @@
         AxisError = type(e)
 
 
+def normalize_to_array(x):
+    if 'cupy' in str(type(x)):
+        return x.get()
+    else:
+        return x
+
+
 def allclose(a, b, equal_nan=False, **kwargs):
+    a = normalize_to_array(a)
+    b = normalize_to_array(b)
```

Comment:
So this is unpleasant.  What are some better ways?  A Dispatch function?

Reply:
I don't see a better way without requiring the cupy dependency, which we presumably don't want to do. cupy does have isnan, and allclose would be easy to do.